pennylane_lightning/core/src/measurements/MeasurementsBase.hpp

```diff
@@ -449,7 +439,42 @@ template <class StateVectorT, class Derived> class MeasurementsBase {
         return outcome_map;
     }
 
+    void set_DeviceSeed(
+        const std::optional<std::size_t> &deviceSeed = std::nullopt) {
+        this->DeviceSeed = deviceSeed;
+    }
+
+    void set_DeviceSeedFromPRNG(std::mt19937 *gen) {
+        if (gen != nullptr) {
+            this->set_DeviceSeed((*gen)());
+        }
+    }
+
+    void activate_DeviceSeed() {
+        if (this->DeviceSeed.has_value()) {
+            setSeed(this->DeviceSeed.value());
+        } else {
+            setRandomSeed();
+        }
+    }
+
   private:
+    /**
+     * @brief Set the seed of the internal random generator
+     *
+     * @param seed Seed
+     */
+    void setSeed(const std::size_t seed) { rng.seed(seed); }
```

Comment:
Is there a reason why we don't use the existing seed setting methods? It is a bit confusing to have two suites of seed setters here. 

What about this: we update the existing `setSeed` to take in an optional seed integer, and depending on the option we either `rng.seed(seed)` or `std::random_device rd; rng.seed(rd())`?

Reply:
You are right. I have replaced them like this:
```
void activateDeviceSeed() {
        std::size_t seed = this->DeviceSeed.has_value()
                               ? this->DeviceSeed.value()
                               : this->generateRandomSeed();
        rng.seed(seed);
    }

  private:
    /**
     * @brief Generate a random seed
     *
     */
    std::size_t generateRandomSeed() {
        std::random_device rd;
        return rd();
    }
```
What do you think?

Reply:
The reason I created a member variable with the seed info is because we need to be able to propagate the seed to newly created derived measures. E.g.:
```
// we are inside the base measure class, which already has a seed
Derived measure(_statevector);
measure.setDeviceSeed(this->DeviceSeed);
return measure.generate_samples(num_shots);
```
`setDeviceSeed` propagates the seed. `measure.generate_samples(...)` will actually apply it to the base rng.

Reply:
We set the value separately from applying it to the rng because not all the simulators will actually use the rng. LQ and LG do use it, but LK only uses the seed, it does not use the rng, it has its own way to generate samples.

Reply:
It did not change drastically. Instead of taking the seed from an argument, it takes it from the `DeviceSeed `member variable. This way, the LK simulator (in fact any simulator) does not need to check if the PRNG is set or not and choose which `generate_samples `signature to call. Instead, it calls the `generate_samples` method takes care of this logic.